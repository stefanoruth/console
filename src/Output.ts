import rl from 'readline'

const colorList = {
	reset: '\x1b[0m',
	cyan: '\x1b[36m',
	yellow: '\x1b[33m',
}

type Colors = keyof typeof colorList

export class Output {
	/**
	 * Write a line to the console.
	 */
	line(message: string, newLine: boolean = false, color: Colors = 'reset'): void {
		process.stdout.write(color + '%s' + colorList.reset, message)

		if (newLine) {
			this.newLine()
		}
	}

	/**
	 * Display a table on the console.
	 */
	table(...args: any[]): void {
		console.table(args)
	}

	/**
	 * Display a set of new lines.
	 */
	newLine(count: number = 1) {
		process.stdout.write('\n'.repeat(count))
	}

	/**
	 * Ask the user for a question.
	 */
	ask(question: string): Promise<string> {
		return new Promise(resolve => {
			const r = rl.createInterface({
				input: process.stdin,
				output: process.stdout,
			})

			r.question(question + '\n', (answer: string) => {
				r.close()
				resolve(answer)
			})
		})
	}

	/**
	 * Ask the user if they really wanna do it.
	 */
	confirm(question: string, defaultValue: boolean = true): Promise<boolean> {
		return new Promise(resolve => {
			const r = rl.createInterface({
				input: process.stdin,
				output: process.stdout,
			})

			r.question(question + '\n', (answer: string) => {
				r.close()
				resolve(answer === 'Y')
			})
		})
	}
}
