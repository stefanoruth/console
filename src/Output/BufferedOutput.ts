export class BufferedOutput {
	protected buffer = ''

	/**
	 * Empties buffer and returns its content.
	 */
	fetch(): string {
		const content = this.buffer
		this.buffer = ''

		return content
	}

	/**
	 * Writes a message to the buffer.
	 */
	protected write(message: string, newline: boolean = false) {
		this.buffer += message

		if (newline) {
			this.buffer += '\n'
		}
	}
}
