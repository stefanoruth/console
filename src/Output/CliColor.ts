export type Colors = 'black' | 'red' | 'green' | 'yellow' | 'blue' | 'meganta' | 'cyan' | 'white'

interface ColorDefinition {
	set: string
	unset: string
}

// https://en.wikipedia.org/wiki/ANSI_escape_code#Colors
function color(set: number, unset: number = 0): ColorDefinition {
	return {
		set: `\x1b[${set}m`,
		unset: `\x1b[${unset}m`,
	}
}

export class CliColor {
	protected foregroundColors: { [c: string]: ColorDefinition } = {
		black: color(30),
		red: color(31),
		green: color(32),
		yellow: color(33),
		blue: color(34),
		magenta: color(35),
		cyan: color(36),
		white: color(37),
	}

	protected backgroundColors: { [c: string]: ColorDefinition } = {
		black: color(40),
		red: color(41),
		green: color(42),
		yellow: color(43),
		blue: color(44),
		magenta: color(45),
		cyan: color(46),
		white: color(47),
	}
}
