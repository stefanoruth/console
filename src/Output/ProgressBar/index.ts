export * from './ProgressBar'
export * from './ProgressCounter'
export * from './ProgressFormat'
export * from './ProgressStyle'
