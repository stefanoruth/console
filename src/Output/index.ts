export * from './ErrorHandler'
export * from './ProgressBar'
export * from './Question'
export * from './Style'
export * from './Table'
export * from './Descriptor'
export * from './Output'
export * from './OutputFormatter'
export * from './Terminal'
export * from './Verbosity'
export * from './Writer'
