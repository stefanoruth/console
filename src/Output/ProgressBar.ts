import { Output } from './Output'
import { Verbosity } from './Verbosity'

export class ProgressBar {
	protected barWidth: number = 28
	protected barChar?: string
	protected emptyBarChar: string = '-'
	protected progressChar: string = '>'
	protected format?: string
	protected internalFormat?: string
	protected redrawFreq: number = 1
	protected step: number = 0
	protected startTime?: number
	protected stepWidth?: number
	protected percent: number = 0.0
	protected formatLineCount?: number
	protected messages: string[] = []
	protected _overwrite: boolean = true
	protected firstRun: boolean = true

	/**
	 * Build new progress bar.
	 */
	constructor(protected output: Output, protected max: number = 0) {
		this.startTime = new Date().getTime()
	}

	/**
	 * Starts the progress output.
	 */
	start(max?: number) {
		this.startTime = new Date().getTime()
		this.step = 0
		this.percent = 0.0

		if (max !== undefined) {
			this.setMaxSteps(max)
		}

		this.display()

		return this
	}

	/**
	 * Advances the progress output X steps.
	 */
	advance(step: number = 1) {
		this.setProgress(this.step + step)
	}

	/**
	 * Sets whether to overwrite the progressbar, false for new line.
	 */
	setOverwrite(overwrite: boolean) {
		this._overwrite = overwrite
	}

	/**
	 * Set the progress output X steps.
	 */
	setProgress(step: number) {
		console.log(step)
		if (this.max && step > this.max) {
			this.max = step
		} else if (step < 0) {
			step = 0
		}

		const prevPeriod = this.step / this.redrawFreq
		const currPeriod = step / this.redrawFreq
		this.step = step
		this.percent = this.max ? this.step / this.max : 0

		if (prevPeriod !== currPeriod || this.max === step) {
			this.display()
		}
	}

	/**
	 * Set max steps.
	 */
	setMaxSteps(max: number) {
		this.format = undefined
		this.max = Math.max.apply(null, [0, max])
		this.stepWidth = this.max ? this.max.toString().length : 4
	}

	/**
	 * Finishes the progress output.
	 */
	finish() {
		if (!this.max) {
			this.max = this.step
		}

		if (this.step === this.max && !this.overwrite) {
			// prevent double 100% output
			return
		}

		this.setProgress(this.max)
	}

	/**
	 * Outputs the current progress string.
	 */
	display(): void {
		if (this.output.getVerbosity() === Verbosity.quiet) {
			return
		}

		if (this.format === null) {
			this.setRealFormat(this.internalFormat || this.determineBestFormat())
		}

		this.overwrite(this.buildLine())
	}

	/**
	 * Removes the progress bar from the current line.
	 *
	 * This is useful if you wish to write some output
	 * while a progress bar is running.
	 * Call display() to show the progress bar again.
	 */
	clear() {
		if (!this.overwrite) {
			return
		}

		if (this.format === null) {
			this.setRealFormat(this.internalFormat || this.determineBestFormat())
		}

		this.overwrite('')
	}

	protected setRealFormat(format: string) {
		// try to use the _nomax variant if available
		if (!this.max) {
			//
		}
		// if (!$this -> max && null !== self:: getFormatDefinition($format.'_nomax')) {
		//     $this -> format = self:: getFormatDefinition($format.'_nomax');
		// } elseif(null !== self:: getFormatDefinition($format)) {
		//     $this -> format = self:: getFormatDefinition($format);
		// } else {
		//     $this -> format = $format;
		// }
		// $this -> formatLineCount = substr_count($this -> format, "\n");
	}

	/**
	 * Overwrites a previous message to the output.
	 */
	protected overwrite(message: string) {
		// if ($this -> overwrite) {
		//     if (!$this -> firstRun) {
		//         if ($this -> output instanceof ConsoleSectionOutput) {
		//             $lines = floor(Helper:: strlen($message) / $this -> terminal -> getWidth()) + $this -> formatLineCount + 1;
		//             $this -> output -> clear($lines);
		//         } else {
		//             // Erase previous lines
		//             if ($this -> formatLineCount > 0) {
		//                 $message = str_repeat("\x1B[1A\x1B[2K", $this -> formatLineCount).$message;
		//             }
		//             // Move the cursor to the beginning of the line and erase the line
		//             $message = "\x0D\x1B[2K$message";
		//         }
		//     }
		// } elseif($this -> step > 0) {
		//     $message = PHP_EOL.$message;
		// }
		// $this -> firstRun = false;
		// $this -> output -> write($message);
	}

	protected determineBestFormat(): string {
		return ''
		// switch ($this -> output -> getVerbosity()) {
		//     // OutputInterface::VERBOSITY_QUIET: display is disabled anyway
		//     case OutputInterface:: VERBOSITY_VERBOSE:
		//         return $this -> max ? 'verbose' : 'verbose_nomax';
		//     case OutputInterface:: VERBOSITY_VERY_VERBOSE:
		//         return $this -> max ? 'very_verbose' : 'very_verbose_nomax';
		//     case OutputInterface:: VERBOSITY_DEBUG:
		//         return $this -> max ? 'debug' : 'debug_nomax';
		//     default:
		//         return $this -> max ? 'normal' : 'normal_nomax';
		// }
	}

	protected static initPlaceholderFormatters(): [] {
		return []
		// return [
		//     'bar' => function (self $bar, OutputInterface $output) {
		//         $completeBars = floor($bar -> getMaxSteps() > 0 ? $bar -> getProgressPercent() * $bar -> getBarWidth() : $bar -> getProgress() % $bar -> getBarWidth());
		//         $display = str_repeat($bar -> getBarCharacter(), $completeBars);
		//         if ($completeBars < $bar -> getBarWidth()) {
		//             $emptyBars = $bar -> getBarWidth() - $completeBars - Helper:: strlenWithoutDecoration($output -> getFormatter(), $bar -> getProgressCharacter());
		//             $display.= $bar -> getProgressCharacter().str_repeat($bar -> getEmptyBarCharacter(), $emptyBars);
		//         }
		//         return $display;
		//     },
		//     'elapsed' => function (self $bar) {
		//         return Helper:: formatTime(time() - $bar -> getStartTime());
		//     },
		//     'remaining' => function (self $bar) {
		//         if (!$bar -> getMaxSteps()) {
		//             throw new LogicException('Unable to display the remaining time if the maximum number of steps is not set.');
		//         }
		//         if (!$bar -> getProgress()) {
		//             $remaining = 0;
		//         } else {
		//             $remaining = round((time() - $bar -> getStartTime()) / $bar -> getProgress() * ($bar -> getMaxSteps() - $bar -> getProgress()));
		//         }
		//         return Helper:: formatTime($remaining);
		//     },
		//     'estimated' => function (self $bar) {
		//         if (!$bar -> getMaxSteps()) {
		//             throw new LogicException('Unable to display the estimated time if the maximum number of steps is not set.');
		//         }
		//         if (!$bar -> getProgress()) {
		//             $estimated = 0;
		//         } else {
		//             $estimated = round((time() - $bar -> getStartTime()) / $bar -> getProgress() * $bar -> getMaxSteps());
		//         }
		//         return Helper:: formatTime($estimated);
		//     },
		//     'memory' => function (self $bar) {
		//         return Helper:: formatMemory(memory_get_usage(true));
		//     },
		//     'current' => function (self $bar) {
		//         return str_pad($bar -> getProgress(), $bar -> getStepWidth(), ' ', STR_PAD_LEFT);
		//     },
		//     'max' => function (self $bar) {
		//         return $bar -> getMaxSteps();
		//     },
		//     'percent' => function (self $bar) {
		//         return floor($bar -> getProgressPercent() * 100);
		//     },
		// ];
	}

	protected static initFormats(): [] {
		return []
		// return [
		//     'normal' => ' %current%/%max% [%bar%] %percent:3s%%',
		//     'normal_nomax' => ' %current% [%bar%]',
		//     'verbose' => ' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%',
		//     'verbose_nomax' => ' %current% [%bar%] %elapsed:6s%',
		//     'very_verbose' => ' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s%',
		//     'very_verbose_nomax' => ' %current% [%bar%] %elapsed:6s%',
		//     'debug' => ' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%',
		//     'debug_nomax' => ' %current% [%bar%] %elapsed:6s% %memory:6s%',
		// ];
	}

	protected buildLine(): string {
		return ''
		// $regex = "{%([a-z\-_]+)(?:\:([^%]+))?%}i";
		// $callback = function ($matches) {
		//     if ($formatter = $this:: getPlaceholderFormatterDefinition($matches[1])) {
		//         $text = $formatter($this, $this -> output);
		//     } elseif(isset($this -> messages[$matches[1]])) {
		//         $text = $this -> messages[$matches[1]];
		//     } else {
		//         return $matches[0];
		//     }
		//     if (isset($matches[2])) {
		//         $text = sprintf('%'.$matches[2], $text);
		//     }
		//     return $text;
		// };
		// $line = preg_replace_callback($regex, $callback, $this -> format);
		// // gets string length for each sub line with multiline format
		// $linesLength = array_map(function ($subLine) {
		//     return Helper:: strlenWithoutDecoration($this -> output -> getFormatter(), rtrim($subLine, "\r"));
		// }, explode("\n", $line));
		// $linesWidth = max($linesLength);
		// $terminalWidth = $this -> terminal -> getWidth();
		// if ($linesWidth <= $terminalWidth) {
		//     return $line;
		// }
		// $this -> setBarWidth($this -> barWidth - $linesWidth + $terminalWidth);
		// return preg_replace_callback($regex, $callback, $this -> format);
	}
}
