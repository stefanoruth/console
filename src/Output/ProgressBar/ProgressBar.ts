import { Output } from '../Output'
import { Terminal } from '../Terminal'
import { ProgressStyle, ProgressFormat } from './ProgressStyle'
import { ProgressCounter } from './ProgressCounter'
import { Verbosity } from '../Verbosity'

export class ProgressBar {
	protected format?: ProgressFormat
	protected style: ProgressStyle
	protected counter: ProgressCounter

	/**
	 * Build new progress bar.
	 */
	constructor(protected output: Output, protected terminal: Terminal) {
		this.counter = new ProgressCounter()
		this.style = new ProgressStyle(this.counter)
	}

	/**
	 * Starts the progress output.
	 */
	start(max?: number) {
		this.counter.start(max)

		this.display()

		return this
	}

	/**
	 * Advances the progress output X steps.
	 */
	advance(step: number = 1) {
		if (this.counter.advance(step)) {
			this.display()
		}
	}

	/**
	 * Set the progress output X steps.
	 */
	setProgress(step: number) {
		if (this.counter.setProgress(step)) {
			this.display()
		}
	}

	/**
	 * Set max steps.
	 */
	setMaxSteps(max: number) {
		this.counter.setMaxSteps(max)
	}

	/**
	 * Finishes the progress output.
	 */
	finish() {
		if (this.counter.finish()) {
			this.display()
		}
	}

	/**
	 * Outputs the current progress string.
	 */
	protected display(): void {
		if (!this.format) {
			this.format = this.getFormat()
		}

		const render = this.style.format(this.format)

		this.terminal.clearLine()
		this.terminal.cursorReset()
		this.terminal.write(render())
	}

	/**
	 * Find out which format we should render for the user.
	 */
	protected getFormat(): ProgressFormat {
		switch (this.output.getVerbosity()) {
			// OutputInterface::VERBOSITY_QUIET: display is disabled anyway
			case Verbosity.verbose:
				return this.counter.getMaxSteps() ? 'verbose' : 'verboseNomax'

			case Verbosity.veryVerbose:
				return this.counter.getMaxSteps() ? 'veryVerbose' : 'veryVerboseNomax'

			case Verbosity.debug:
				return this.counter.getMaxSteps() ? 'debug' : 'debugNomax'

			default:
				return this.counter.getMaxSteps() ? 'normal' : 'normalNomax'
		}
	}
}
