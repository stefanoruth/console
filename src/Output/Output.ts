import rl from 'readline'
import { Color, CliColor } from './CliColor'
import { ProgressBar } from './ProgressBar'
import { Table } from './Table'
import { Question } from './Question/Question'
import { Writer } from './Writer'

export class Output {
	constructor(public writer: Writer = new Writer(), protected color: CliColor = new CliColor()) {}

	/**
	 * Write a line to the console.
	 */
	line(message: string, newLine: boolean = true, color?: Color) {
		this.writer.write(this.color.apply(message, { text: color }), newLine)
	}

	/**
	 * Formats a command title.
	 */
	title(message: string) {
		//
	}

	/**
	 * Formats a section title.
	 */
	section(message: string) {
		//
	}

	/**
	 * Formats a list.
	 */
	listing(elements: any[]) {
		//
	}

	/**
	 * Formats a command comment.
	 */
	comment(message: string | string[]) {
		this.writer.block(message, undefined, undefined, '<fg=default;bg=default> // </>', false, false)
	}

	/**
	 * Formats a success result bar.
	 */
	success(message: string) {
		this.writer.writeln(this.color.apply(message, { text: 'green' }))
	}

	/**
	 * Formats an error result bar.
	 */
	error(message: string) {
		this.writer.writeln(this.color.apply(message, { text: 'white', bg: 'red' }))
		// this.block(message, 'ERROR', 'fg=white;bg=red', ' ', true)
	}

	/**
	 * Formats an warning result bar.
	 */
	warning(message: string) {
		this.writer.writeln(this.color.apply(message, { text: 'black', bg: 'yellow' }))
		// this.block(message, 'WARNING', 'fg=black;bg=yellow', ' ', true)
	}

	/**
	 * Formats a note admonition.
	 */
	note(message: string) {
		this.writer.writeln(this.color.apply(message, { text: 'yellow' }))
		// this.block(message, 'NOTE', 'fg=yellow', ' ! ')
	}

	/**
	 * Formats a caution admonition.
	 */
	caution(message: string) {
		this.writer.writeln(this.color.apply(message, { text: 'white', bg: 'red' }))
		// this.block(message, 'CAUTION', 'fg=white;bg=red', ' ! ', true)
	}

	/**
	 * Display a table on the console.
	 */
	table(rows: object[], columns?: object) {
		const table = new Table(this)
		table.setHeaders(columns)
		table.setRows(rows)

		table.render()
		this.newLine()
	}

	/**
	 * Display a set of new lines.
	 */
	newLine(count: number = 1) {
		this.writer.newLine(count)
	}

	/**
	 * Ask the user for a question.
	 */
	ask(question: string) {
		return new Question().ask(question)
	}

	askQuestion(question: Question) {
		//
	}

	/**
	 * Ask the user if they really wanna do it.
	 */
	confirm(question: string, defaultValue: boolean = true): Promise<boolean> {
		return new Promise(resolve => {
			const r = rl.createInterface({
				input: process.stdin,
				output: process.stdout,
			})

			r.question(question + '\n', (answer: string) => {
				r.close()
				resolve(answer.toUpperCase() === 'Y')
			})
		})
	}

	/**
	 * Start a new progress bar
	 */
	progressBar(max: number = 0): ProgressBar {
		const progressBar = new ProgressBar(this, max)

		progressBar.start()

		return progressBar
	}
}
