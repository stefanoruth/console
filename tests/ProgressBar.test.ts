import { ProgressBar, ProgressCounter, ProgressFormat, ProgressStyle } from '../src/Output/ProgressBar'
import { Mock } from 'ts-mockery'

describe('ProgressBar', () => {
	describe('Counter', () => {
		test('Automatic start timer', () => {
			const c = new ProgressCounter()

			expect(() => c.getStartTime()).toThrow()
			c.start()
			expect(() => c.getStartTime()).not.toThrow()
		})

		test('Works with a max value', () => {
			const c = new ProgressCounter()

			expect(c.start(200).getMaxSteps()).toBe(200)
		})

		test('Works without a max value', () => {
			const c = new ProgressCounter()

			expect(c.start().getMaxSteps()).toBe(0)
		})

		test('Need to call start before advancing', () => {
			const c = new ProgressCounter()

			expect(() => c.advance()).toThrow()
			c.start()
			expect(() => c.advance()).not.toThrow()
		})

		test('Need to call start before finishing', () => {
			const c = new ProgressCounter()

			expect(() => c.finish()).toThrow()
			c.start()
			expect(() => c.finish()).not.toThrow()
		})

		test('Can calcuate progress', () => {
			const c = new ProgressCounter()

			c.start(5)

			c.advance()
			expect(c.getProgress()).toBe(1)
			expect(c.getProgressPercent()).toBe(0.2)

			c.advance(3)
			expect(c.getProgress()).toBe(4)
			expect(c.getProgressPercent()).toBe(0.8)
		})

		test('Finish set step to max value', () => {
			const c = new ProgressCounter()

			c.start(5).advance()
			expect(c.getProgress()).toBe(1)

			c.finish()
			expect(c.getProgress()).toBe(5)
		})

		test('Can advance beyond the max level', () => {
			const c = new ProgressCounter()
			c.start(5)

			expect(c.getMaxSteps()).toBe(5)
			c.advance(10)
			expect(c.getMaxSteps()).toBe(10)
		})

		test('Can advance backwards but not beyond zero', () => {
			const c = new ProgressCounter()
			c.start()

			c.advance(-1)
			expect(c.getProgress()).toBe(0)

			c.advance(10)
			expect(c.getProgress()).toBe(10)

			c.advance(-5)
			expect(c.getProgress()).toBe(5)

			c.advance(-10)
			expect(c.getProgress()).toBe(0)
		})
	})

	describe('Style', () => {
		test('Calcuate elapsed time', () => {
			const run = (dateFn: (d: Date) => number) => {
				const c = Mock.of<ProgressCounter>({ getStartTime: () => dateFn(new Date()) })
				const s = new ProgressStyle(c)

				return s.elapsed().trim()
			}

			// expect(run(d => d.setDate(d.getDate() - 1))).toBe('1 day')
			// expect(run(d => d.setHours(d.getHours() - 1))).toBe('1 hr')
		})

		test('Step widths', () => {
			const run = (max?: number) => {
				const c = new ProgressCounter()

				if (max) {
					c.setMaxSteps(max)
				}

				return new ProgressStyle(c).getStepWidth()
			}

			expect(run()).toBe(4)
			expect(run(1)).toBe(1)
			expect(run(99)).toBe(2)
			expect(run(200)).toBe(3)
			expect(run(50000)).toBe(5)
		})

		test('Display max step', () => {
			const run = (max?: number) => {
				const c = new ProgressCounter()

				if (max) {
					c.setMaxSteps(max)
				}

				return new ProgressStyle(c).max()
			}

			expect(run()).toBe('0')
			expect(run(50)).toBe('50')
		})
	})
})
