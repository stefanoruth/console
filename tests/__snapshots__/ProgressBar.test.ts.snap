// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Render Format 1`] = `"20/80 [=======>--------------------]  25%"`;

exports[`ProgressBar Render Format 2`] = `"  20 [====================>-------]"`;

exports[`ProgressBar Render Format 3`] = `"20/80 [=======>--------------------]  25% < 1 sec"`;

exports[`ProgressBar Render Format 4`] = `"  20 [====================>-------] < 1 sec"`;

exports[`ProgressBar Render Format 5`] = `"20/80 [=======>--------------------]  25% < 1 sec/< 1 sec"`;

exports[`ProgressBar Render Format 6`] = `"  20 [====================>-------] < 1 sec"`;

exports[`ProgressBar Render Format 7`] = `"20/80 [=======>--------------------]  25% < 1 sec/< 1 sec  500 B"`;

exports[`ProgressBar Render Format 8`] = `"  20 [====================>-------] < 1 sec  500 B"`;

exports[`ProgressBar Render Format 9`] = `""`;

exports[`ProgressBar Render Format 10`] = `""`;
